docs/sphinx/quickstart.rst

```diff
@@ -53,59 +54,107 @@ versions, is available at :ref:`prerequisites`.
 Building |hpx|
 ==============
 
-Once you have the source code and the dependencies, set up a separate build
-directory and configure the project. Assuming all your dependencies are in paths
+Once you have the source code and the dependencies and assuming all your dependencies are in paths
 known to |cmake|, the following gets you started:
 
-.. code-block:: sh
+#. First, set up a separate build directory to configure the project:
 
-    # In the HPX source directory
-    mkdir build && cd build
-    cmake -DCMAKE_INSTALL_PREFIX=/install/path ..
-    make install
+  .. code-block:: sh
 
-This will build the core |hpx| libraries and examples, and install them to your
-chosen location. If you want to install |hpx| to system folders, simply leave out
-the ``CMAKE_INSTALL_PREFIX`` option. This may take a while. To speed up the
-process, launch more jobs by passing the ``-jN`` option to ``make``.
+      cd ..
```

Comment:
`build` can actually also be inside the source directory, and we even ignore it in `.gitignore`. I for example have a `~/src/` directory with all my cloned repos, and if I put `build` inside the source directory for all of those I can just call them all `build`. If I go up to the source directory I'd have to start calling them `hpx-build` etc. Do you think it's clearer to have the build directory completely outside the source directory? It might also be a good idea to just add a (bash) comment at the top of the code block saying in which directory you assume the user is in at the moment?

Reply:
Which do you think is the most proper way to do it? Having `build` inside or outside the source directory? Since there is always another way to do it, I think we can chose the optimal way for the user, so that they would just copy-paste the commands without wondering which directory is the proper one. What do you think?

Reply:
I think having a `build` directory inside the source directory is a very common convention, but obviously not universal. At the very least having `build` inside the source directory is pretty much guaranteed to not conflict with a `build` directory that the user may have created in the parent directory for other purposes. I'd keep it inside the source directory.